tapchannel/aux_invoice_manager.go

```diff
@@ -0,0 +1,197 @@
+
+		// Start the interception in its own goroutine.
+		s.Wg.Add(1)
+		go func() {
+			defer s.Wg.Done()
+
+			ctx, cancel := s.WithCtxQuitNoTimeout()
+			defer cancel()
+
+			err := s.cfg.InvoiceHtlcInterceptor.HtlcInterceptor(
+				ctx, s.handleInvoiceAccept,
+			)
+			if err != nil {
+				log.Errorf("Error setting up invoice "+
+					"acceptor: %v", err)
+				return
+			}
+		}()
+	})
+	return startErr
+}
+
+func (s *AuxInvoiceManager) handleInvoiceAccept(_ context.Context,
+	req lndclient.InterceptedInvoiceHtlcRequest) (
+	*lndclient.InterceptedInvoiceHtlcResponse, error) {
+
+	// By default, we'll return the same amount that was requested.
+	resp := &lndclient.InterceptedInvoiceHtlcResponse{
+		CircuitKey: req.CircuitKey,
+		AmtPaid:    req.ExitHtlcAmt,
+	}
+
+	jsonBytes, err := taprpc.ProtoJSONMarshalOpts.Marshal(req.Invoice)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode response: %w", err)
+	}
+
+	log.Debugf("Received invoice: %s", jsonBytes)
+	log.Debugf("Received wire custom records: %v",
+		limitSpewer.Sdump(req.WireCustomRecords))
+
+	// No custom record on the HTLC, so we have nothing to do.
+	if len(req.WireCustomRecords) == 0 {
+		return resp, nil
+	}
+
+	htlcBlob, err := req.WireCustomRecords.Serialize()
+	if err != nil {
+		return nil, fmt.Errorf("error serializing custom records: %w",
+			err)
+	}
+
+	htlc, err := rfqmsg.DecodeHtlc(htlcBlob)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode htlc: %w", err)
+	}
+
+	log.Debugf("Received htlc: %v", limitSpewer.Sdump(htlc))
+
+	// If we don't have an RFQ ID, then this is likely a keysend payment,
+	// and we don't modify the amount (since the invoice amount will match
+	// the HTLC amount).
+	if htlc.RfqID.ValOpt().IsNone() {
+		return resp, nil
+	}
+
+	rfqID := htlc.RfqID.ValOpt().UnsafeFromSome()
+	acceptedQuotes := s.cfg.RfqManager.PeerAcceptedBuyQuotes()
+	quote, ok := acceptedQuotes[rfqID.Scid()]
+	if !ok {
+		return nil, fmt.Errorf("no accepted quote found for RFQ SCID "+
+			"%d", rfqID.Scid())
+	}
+
+	log.Debugf("Found quote for SCID %d: %#v", rfqID.Scid(), quote)
+
+	mSatPerAssetUnit := quote.AskPrice
+	htlcAssetAmount := lnwire.MilliSatoshi(htlc.Amounts.Val.Sum())
+	resp.AmtPaid = htlcAssetAmount * mSatPerAssetUnit
+
+	// If all previously accepted HTLC amounts plus the intercepted HTLC
+	// amount together add up to just about the asset invoice amount, then
+	// we can settle the HTLCs to address the rounding error.
+	var acceptedHtlcSum lnwire.MilliSatoshi
+	for _, invoiceHtlc := range req.Invoice.Htlcs {
+		acceptedHtlcSum += lnwire.MilliSatoshi(invoiceHtlc.AmtMsat)
+	}
+
+	// We assume that each shard can have a rounding error of up to 1 asset
+	// unit. So we allow the final amount to be off by up to 1 asset unit
+	// per accepted HTLC (plus the one we're currently processing).
+	allowedMarginAssetUnits := lnwire.MilliSatoshi(
+		len(req.Invoice.Htlcs) + 1,
+	)
+	allowedMarginMSat := allowedMarginAssetUnits * mSatPerAssetUnit
```

Comment:
Yeah, correct. But I don't know what else to do at this point. Especially if the sender is BTC only, all they see is an amount in satoshi which they are allowed to split up. And they aren't even aware of the issue that making more splits increases the inaccuracy.
But with the current implementation the error gets smaller with the precision of the asset. So if the asset is one hundredth of a cent, then even with 16 parts (which is the current default maximum for MPP) you only get an error rate of 0.16 cents, which IMO is fine.

Reply:
Yeah it's not something we should deal with atm. Let's advocate for precision and small value units.
Could open an issue for this so that we revisit, for now LGTM

Reply:
Sounds good to me. Thanks for creating [the issue](https://github.com/lightninglabs/taproot-assets/issues/920)!